src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -1170,6 +1179,190 @@ def corner2offset(self, block, corner):
 
         return offset
 
+#===============================================================================
+#
+class SchedulerYarn(AgentSchedulingComponent):
+
+    # FIXME: clarify what can be overloaded by Scheduler classes
+
+    # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg):
+
+        AgentSchedulingComponent.__init__(self, cfg)
+
+    # --------------------------------------------------------------------------
+    #
+    def _configure(self):
+
+        try:
+            #-----------------------------------------------------------------------
+            # Find out how many applications you can submit to YARN. And also keep
+            # this check happened to update it accordingly
+
+
+            #if 'rm_ip' not in self._cfg['lrms_info']:
+            #    raise RuntimeError('rm_ip not in lm_info for %s' \
+            #            % (self.name))
+
+            self._log.info('Checking rm_ip %s' % self._cfg['lrms_info']['lm_info']['rm_ip'])
+            self._rm_ip = self._cfg['lrms_info']['lm_info']['rm_ip']
+            self._service_url = self._cfg['lrms_info']['lm_info']['service_url']
+            self._rm_url = self._cfg['lrms_info']['lm_info']['rm_url']
+            self._client_node = self._cfg['lrms_info']['lm_info']['nodename']
+
+            sample_time = rpu.timestamp()
+            yarn_status = ul.urlopen('http://{0}:8088/ws/v1/cluster/scheduler'.format(self._rm_ip))
+
```

Comment:
Except from the actual URL everything else is standard in order to access the status of the YARN scheduler. Maybe, the port can be parametrized, but I have to check first.